// @ionic/cli/commands/state.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StateCommand = void 0;
const cli_framework_output_1 = require("@ionic/cli-framework-output");
const format_1 = require("@ionic/cli-framework/utils/format");
const color_1 = require("../lib/color");
const command_1 = require("../lib/command");
const errors_1 = require("../lib/errors");
class StateCommand extends command_1.Command {
    async getMetadata() {
        return {
            name: 'state',
            type: 'global',
            summary: '',
            groups: ["hidden" /* HIDDEN */],
        };
    }
    async run() {
        const data = [
            [`${cli_framework_output_1.indent(4)}${color_1.input('ionic cordova platform save')}`, `save existing installed platforms to ${color_1.strong('config.xml')}`],
            [`${cli_framework_output_1.indent(4)}${color_1.input('ionic cordova plugin save')}`, `save existing installed plugins to ${color_1.strong('config.xml')}`],
            [`${cli_framework_output_1.indent(4)}${color_1.input('ionic cordova platform --help')}`, `view help page for managing Cordova platforms`],
            [`${cli_framework_output_1.indent(4)}${color_1.input('ionic cordova plugin --help')}`, `view help page for managing Cordova plugins`],
            [`${cli_framework_output_1.indent(4)}${color_1.input('ionic cordova prepare')}`, `install platforms and plugins listed in ${color_1.strong('config.xml')}`],
        ];
        throw new errors_1.FatalException(`${color_1.input('ionic state')} has been removed.\n\n` +
            `We recommend using Cordova directly to manage Cordova plugins and platforms.\n` +
            `The following commands fulfill the old ${color_1.input('ionic state')} functionality:\n\n` +
            `${format_1.columnar(data, {})}\n\n` +
            `See ${color_1.strong('https://cordova.apache.org/docs/en/latest/platform_plugin_versioning_ref/')} for detailed information.\n`);
    }
}
exports.StateCommand = StateCommand;
